### ISSUE TITLE
Persistent Spell Damage buffs in non-PLAY zones

### ISSUE THREAD

**Issue by Xinhuan** [HearthSim/hs-bugs#347]
http://us.battle.net/hearthstone/en/forum/topic/20419282159

When using Brann Bronzebeard + Faceless Manipulator on a minion that is buffed with an enchantment that grants Spell Damage, the double copy/transform caused by Brann causes the first transformed copy of the minion to be moved to the SETASIDE zone where the copied enchantment remains active permanently for the rest of the game. This means that the player receives the effects of Spell Damage permanently even though the minion is not in the Battlefield.

In the following video, we show this bug with Velen's Chosen and Ancient Mage.

Video: https://www.youtube.com/watch?v=S3bUuC7Ez5Y
Log: https://gist.github.com/Xinhuan/fb2cdfabd0abf5793866

There is no bug on minions that directly give Spell Damage, the bug only exists on enchantment based Spell Damage. Dalaran Aspirant is not tested but is assumed to be bugged the same way.

For additional reference, Velen's Chosen when cast on a target that is in the DECK zone (Malorne) also causes persistent Spell Damage until Malorne is drawn from the deck: https://www.youtube.com/watch?v=ceQGYDdzdUA&t=6m30s

There might be other enchantments that can give visible side effects other than Spell Damage, but I can't think of any right now.

**Comment by Patashu** [comment]
Note that this bug occurs because enchantments created and attached to minions not in play are themselves considered in play, erroneously, and thus have all the effects and triggers and update relate to being in play. (Although it's not clear what the intended behaviour should be in general, such as with Shadow Madness)

**Comment by Xinhuan** [comment]
Well enchantments that are created and attached to minions in the Hand are probably considered in play to allow for mana discounts, and or buffs via Mistcaller/Bolvar and have them be visibly shown that they are indeed in effect. It would be extremely hard to count how many minions died if Bolvar was not shown in hand with his buff active.

Only buffs with side effects are Spell Damage, Shadow Madness, Nightmare, and Corruption.... I think, and we have a video with Shadow Madness stealing cards in deck, as well as in hand (the Malorne video earlier).

**Comment by Patashu** [comment]
Report in 4.2:

http://us.battle.net/hearthstone/en/forum/topic/20742875769

But interestingly, this report only makes sense if death processing happens between the Targeting Phase and the Spell Text Phase (and in 4.1 and earlier it did not). So I would love to reproduce this to make sure it works as described.

**Comment by 10925** [comment]
https://www.youtube.com/watch?v=hBtYp0tQa6I
https://www.youtube.com/watch?v=Zxjo1IddHl8
https://www.youtube.com/watch?v=kuxYV0qupu0
Seems Like they add death processing the Targeting Phase and the Spell Text Phase, and spell bender was no longer trigger in Targeting Phase……

**Comment by Patashu** [comment]
10925, thank you for your videos. So it seems that there are now a Targeting Phase and a Spellbender Phase in that order, and death processing doesn't occur after the Spellbender Phase but does after the Targeting Phase. I've revised the advanced rulebook to reflect this new understanding.

Does this mean that Blizzard feels that Spellbender and Eydis/etc are not the same kind of trigger and thus shouldn't occur in the same Phase? Or is it a bug?

**Comment by jleclanche** [comment]
@Patashu It's not necessarily a new phase, it could simply be a bug with how the event is broadcast. Eg. the event is not updated when the retargeting happens, thus reaching eydis.

**Comment by PattuX** [comment]
Still happening in 5.0

To file this here correctly as well, Eydis, Fjola and Dragonkin Sorcerer moved to the on-play phase of playing a spell, making Spellbender have its own phase.

* https://www.youtube.com/watch?v=IjU_KzLmv7s

**Comment by VoidMayonnaise** [comment]
https://youtu.be/ewkaBZvwZo0
Already fixed.

**Comment by VoidMayonnaise** [comment]
Seems that this bug appears again, since Malorne and Weasel Tunneler's deathrattles are actually move itself into its owner's setaside zone and shuffle a copy into the deck, so if you play Velen's Chosen on a friendly Malorne or Weasel Tunneler and kill it before the spell text phase, the buff will attach to the one in the setaside zone, and you'll acquire persistent Spell Damage.
Maybe Sherazin has the same interaction.
https://youtu.be/7uwy1F54wA0
(This one is tested in Patch 8.2)
https://youtu.be/80wCxI_v1PI
(I retest in Patch 9.0 by Weasel Tunneler)

**Comment by PattuX** [comment]
Maybe they only fixed the interaction for Brann + Faceless? Seems like we were a bit lazy here only testing one way to reproduce this bug.

**Comment by troggnostupidhs** [comment]
Is this the same bug?

https://www.reddit.com/r/hearthstone/comments/a3vk9v/bug_immortal_prelate_will_give_you_spell_damage/

> Bug: Immortal Prelate will give you Spell Damage, no matter where it is (Hand, deck, etc.)

![image](https://user-images.githubusercontent.com/22508641/49629104-c1c35180-f9b5-11e8-8592-404ca1adb330.png)

> Here is the proof that it is not just a visual bug. It does actually deal 14 damage

![image](https://user-images.githubusercontent.com/22508641/49629114-cee04080-f9b5-11e8-94c1-31de36d1f0e8.png)

**Comment by PattuX** [comment]
Yes, the reason this happens is the same. The minion may not be in play, however the enchantment always is.

**Comment by troggnostupidhs** [comment]
Immortal Prelate is fixed. Does anyone want to do any other tests for this?

![hearthstone screenshot 12-10-18 19 35 40](https://user-images.githubusercontent.com/22508641/49770551-fd5e6400-fcb2-11e8-944a-47434b0f0dd8.png)
![hearthstone screenshot 12-10-18 19 35 55](https://user-images.githubusercontent.com/22508641/49770552-fd5e6400-fcb2-11e8-8649-a660c85744c2.png)

**Comment by troggnostupidhs** [comment]
Need to test more.

**Comment by troggnostupidhs** [comment]
![hearthstone screenshot 12-11-18 09 44 30](https://user-images.githubusercontent.com/22508641/49808038-8c0dc800-fd29-11e8-830c-f0166e63442d.png)
![hearthstone screenshot 12-11-18 09 44 38](https://user-images.githubusercontent.com/22508641/49808039-8c0dc800-fd29-11e8-8f98-2b60085f7e9c.png)
![hearthstone screenshot 12-11-18 09 44 56](https://user-images.githubusercontent.com/22508641/49808040-8ca65e80-fd29-11e8-9719-3d9eaac3d71b.png)

AAEBAa0GAv4D7QUOswH7AZcCnALlBIUIsQiPD/wPt2yErQKxsALiygKm8AIA

**Comment by troggnostupidhs** [comment]
https://us.battle.net/forums/en/hearthstone/topic/20770596920#post-7

> Yes, that is correct. It looks like this didn't get included in the Hotfix notes. Sorry about that. Thanks for following up and yes the spell damage has been fixed.

**Comment by troggnostupidhs** [comment]
https://us.battle.net/forums/en/hearthstone/topic/20761086790?page=2#post-26

> Spell damage will no longer persist when Immortal Prelate is sent to your deck.

**Comment by Patashu** [comment]
Looks like this might be back?

[2:26 PM] Oura: ...lol Blizzard
[2:26 PM] Oura: The Reno Hero for Mage that makes all minions "disappear"
[2:27 PM] Oura: It doesn't destroy them, so minions that granted spell power continue to do so even after being "disappeared"
[2:29 PM] Oura: Was just watching a Hobbs highlight so maybe it's fixed now
[2:32 PM] Patashu: What minion was giving spell damage +1?
[2:33 PM] Patashu: Was it like ancient mage or something where it's conditional
[2:33 PM] Oura: I'd have to re-watch but I think it was one of the new weird cards that adds spell damage at the end of turn
2:34 PM] Oura: Azerite Elemental

**Comment by Jetz72** [comment]
Seen a couple reports but they were still in my "to confirm" mountain. Probably enough of them by now to say it's real. Does Reno send disappeared minions to the set-aside zone instead of the graveyard? The underlying issue was probably never truly fixed.

**Comment by mawilek** [comment]
there is a replay, if that helps
https://hsreplay.net/replay/NPTX9F9uCnLMrFZdrPnvFP
